internal/auth/password/rewrapping.go

```diff
@@ -0,0 +1,78 @@
+package password
+
+import (
+	"context"
+
+	"github.com/hashicorp/boundary/internal/db"
+	"github.com/hashicorp/boundary/internal/errors"
+	"github.com/hashicorp/boundary/internal/kms"
+)
+
+func init() {
+	kms.RegisterTableRewrapFn("auth_password_argon2_cred", argon2ConfigRewrapFn)
+}
+
+type accountAndScope struct {
+	scopeId   string
+	accountId string
+}
+
+func argon2ConfigRewrapFn(ctx context.Context, dataKeyVersionId string, reader db.Reader, writer db.Writer, kmsRepo *kms.Kms) error {
+	const op = "password.argon2ConfigRewrapFn"
+	repo, err := NewRepository(reader, writer, kmsRepo)
+	if err != nil {
+		return errors.Wrap(ctx, err, op)
+	}
+	rows, err := repo.reader.Query(
+		ctx, `
+select distinct acct.scope_id, acct.public_id
+from auth_password_account acct
+inner join auth_password_argon2_cred cred
+on acct.public_id=cred.password_account_id
+where cred.key_id=?
+`,
```

Comment:
Do you plan on refactoring inline queries like this to use a `const` that is defined in a `query.go` file?

Reply:
Yes